src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,114 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MIN_ARRAY_SIZE = 10;
```

Comment:
Good job on defining a constant for the minimum array size. This improves code readability and maintainability.